pandas/core/frame.pyi

```diff
@@ -897,7 +897,7 @@ class DataFrame(NDFrame):
         normalize: _bool = ...,
         fill_value: Optional[Scalar] = ...,
     ) -> DataFrame: ...
-    def astype(self, dtype: Union[_str, Dict[_str, _str]], copy: _bool = ..., errors: _str = ...,) -> DataFrame: ...
+    def astype(self, dtype, copy: _bool = ..., errors: _str = ...,) -> DataFrame: ...
```

Comment:
This would make the type `Unknown`. Do you want `Any`?

Why not add the extra Dict type used in #32?

Reply:
> This would make the type `Unknown`. Do you want `Any`?
> 
> Why not add the extra Dict type used in #32?

I'm following the pattern used in `pandas/core/generic.pyi` : https://github.com/microsoft/python-type-stubs/blob/e0d426736aed26bef81678d71b7a6aeb8cf436de/pandas/core/generic.pyi#L294

Now, `pandas/core/series.pyi` has this:
https://github.com/microsoft/python-type-stubs/blob/e0d426736aed26bef81678d71b7a6aeb8cf436de/pandas/core/series.pyi#L432-L437

but I couldn't get the `S1` reference to work in `frame.pyi` .

Reply:
Hm, I see. I guess @gramster would have a stronger opinion here.

Reply:
I think we need to add _Dtype as an option (including inside the dict). Let me try that out.

Reply:
Yes, this works:

```
    def astype(self, dtype: Union[Dtype, Dict[_str, Dtype]], copy: _bool = ..., errors: _str = ...,) -> DataFrame: 
```

Irv, do you want to update your PR, or I can make a new PR.

Reply:
I'm thinking it might be better to use:

```
def astype(self, dtype: Union[_str, Dtype, Dict[_str, Union[_str, Dtype]]], copy: _bool = ..., errors: _str = ...,) -> DataFrame: 
```

as `Dtype` only worked because it currently includes `object` but that is a temporary anomaly that may go away once numpy has proper types.

Reply:
> Irv, do you want to update your PR, or I can make a new PR. It's actually easier for me as I will bundle in some other fixes but if you want to show up as a contributor happy to leave it to you.

Probably best if you update it.  Do you have some way of testing changes?